File: CreationalPatterns/Singleton/TypeScript/Singleton.ts
export class Singleton {
    private static instance: Singleton;
    private constructor() {
    }

    public static getInstance(): Singleton {
        if (!Singleton.instance) {
            Singleton.instance = new Singleton();
        }
        return Singleton.instance;
    }

    public someBusinessLogic(): void {
        console.log('Executando lógica de negócio do Singleton!');
    }
}

File: CreationalPatterns/Singleton/Go/singleton.go
package main

import (
	"fmt"
	"sync"
)

type singleton struct {
}

func (s *singleton) SomeBusinessLogic() {
    fmt.Println("Executando lógica de negócio do Singleton!")
}

var instance *singleton
var once sync.Once

func GetInstance() *singleton {
    once.Do(func() {
        instance = &singleton{}
    })
    return instance
}